File: aurora-scala/src/main/scala/net/gree/aurora/scala/domain/datasource/DataSourceIdImpl.scala
package net.gree.aurora.scala.domain.datasource

import net.gree.aurora.domain.datasource.{DataSourceId => JDataSourceId}

private[domain]
case class DataSourceIdImpl(underlying: JDataSourceId) extends DataSourceId {

  def value = underlying.getValue

  override def toString = Seq(s"value = $value").mkString("DataSourceId(",", ", ")")

}

File: aurora-core/src/main/java/net/gree/aurora/domain/datasource/DataSourceIdImpl.java
package net.gree.aurora.domain.datasource;

import org.sisioh.dddbase.model.impl.AbstractIdentity;

import java.util.UUID;

final class DataSourceIdImpl extends AbstractIdentity<UUID> implements DataSourceId {

    DataSourceIdImpl(UUID value) {
        super(value);
    }

    public String toString() {
        return String.format("DataSourceId(value = %s)", value);
    }

}
